<launch>

  <!-- Camera ID -->
  <arg name="camera_id" default="" />

  <!-- Camera driver -->
  <arg name="camera_name" default="$(arg camera_id)" />
  <include file="$(find detection)/launch/$(env KINECT_DRIVER).launch"> 
    <arg name="camera" value="$(arg camera_name)" />
  </include>

  <arg name="intermediate_topic" value="/ground_based_people_detector_$(arg camera_name)/detections" />

  <!-- Launch ground based people detection node -->
  <node pkg="detection" type="ground_based_people_detector" name="ground_based_people_detector_$(arg camera_name)" output="screen" required="true">
    <rosparam command="load" file="$(find detection)/conf/ground_based_people_detector.yaml" /> 
    <param name="classifier_file" value="$(find detection)/data/HogSvmPCL.yaml"/>
    <param name="pointcloud_topic" value="/$(arg camera_name)/depth_registered/points"/>
    <param name="output_topic" value="$(arg intermediate_topic)"/>
    <param name="camera_info_topic" value="/$(arg camera_name)/rgb/camera_info"/>
    <param name="rate" value="60.0"/>  
  </node>

  <!-- Launch HaarDispAda Node-->
  <node name="HaarDispAdaNode_$(arg camera_name)" pkg="detection" type="HaarDispAda174" >	
    <rosparam command="load" file="$(find detection)/conf/haar_disp_ada_detector.yaml" /> 
    <remap from="Color_Image" to="/$(arg camera_name)/rgb/image_color"/>	
    <remap from="Disparity_Image" to="/$(arg camera_name)/depth_registered/disparity"/>
    <remap from="input_detections" to="$(arg intermediate_topic)"/>
    <param name="classifier_file" value="$(find detection)/data/HaarDispAda.xml"/>
  </node>  

</launch>
